src/vacuum_map_parser_ijai/beautify_min.py

```diff
@@ -0,0 +1,1158 @@
+# STRAIGHT REWRITE FROM MI HOME PLUGIN`S CODE
+import base64
+
+class Point(object):
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def __repr__(self) -> str:
+        return f'Point({self.x},{self.y})'
+
+
+class BeautifyMap:
+    def __init__(self):
+        self.map = []
+        self.tRect = {
+            "x": 0,
+            "y": 0,
+            "width": 0,
+            "height": 0
+        }
+        self.x_min = 0
+        self.x_max = 0
+        self.y_min = 0
+        self.y_max = 0
+        self.resolution = 0
+        self.size_x = 0
+        self.size_y = 0
+
+    def setSize(self, x_min1, x_max1, y_min1, y_max1, resolution1):
+        self.map = []
+        self.size_x = 0
+        self.size_y = 0
+        self.x_min = int(resolution1 * round(x_min1 / resolution1))
+        self.y_min = int(resolution1 * round(y_min1 / resolution1))
+        self.x_max = int(resolution1 * round(x_max1 / resolution1))
+        self.y_max = int(resolution1 * round(y_max1 / resolution1))
+        self.size_x = int(round((x_max1 - x_min1) / resolution1))
+        self.size_y = int(round((y_max1 - y_min1) / resolution1))
+        self.resolution = resolution1
+
+    def setMap(self, tMapStruct):
+        self.setSize(tMapStruct.mapHead.minX, tMapStruct.mapHead.maxX,
+                     tMapStruct.mapHead.minY, tMapStruct.mapHead.maxY, tMapStruct.mapHead.resolution)
+
+        mapdata_arr = tMapStruct.mapData.mapData
+
+        tempArray = [0] * len(mapdata_arr)
+
+        for i in range(len(mapdata_arr)):
+            if (mapdata_arr[i] > 127):
```

Comment:
self.map = [-128 if mapdatum > 127 else mapdatum for mapdatum in mapdata_arr] ?

Reply:
Somehow.. yes it does, many direct calls to the mapData property significantly worsens performance.